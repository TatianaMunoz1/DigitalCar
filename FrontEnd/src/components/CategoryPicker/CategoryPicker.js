import React, { useState, useEffect } from "react";
import styles from "./CategoryPicker.module.css";
import axios from "axios";
import Url from "../../Url"

function CategoryPicker(props) {
  const [categories, setCategories] = useState([]);
  const [categoryId, setCategoryId] = useState(null);

  useEffect(() => {
    axios
      .get(Url+"/categories")
      .then((response) => {
        setCategories(response.data);
      })
      .catch((error) => {});
  }, []);

  useEffect(() => {
    props.handleCategoryPicker(categoryId);
  }, [categoryId]);

  const categoryOptions = categories.map((category) => (
    <option
      key={`${category.id}_${category.title}`}
      id={category.id.parseInt}
      value={category.id}
    >
      {category.title}
    </option>
  ));

  return (
    <div className={styles.category}>
      <select name="categories" onChange={(e) => setCategoryId(e.target.value)}>
        <option
          selected
          value=""
          disabled
          hidden
          className={styles.placeholder}
        >
          Seleccione una categor√≠a
        </option>
        {categoryOptions}
      </select>
    </div>
  );
}

export default CategoryPicker;
