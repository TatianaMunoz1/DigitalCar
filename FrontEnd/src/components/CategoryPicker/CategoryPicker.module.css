select {
  color: #000000;
  height: 30px;
  width: 100%;
  font-size: 16px;
  background: #ffffff;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
  border-radius: 5px;
  padding-left: 8px;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}
select:focus {
  outline: 1px solid #ff7e19;
  border: 1px solid #ff7e19;
}
select::-ms-expand {
  display: none;
}
select:not(:-internal-list-box) {
  background-color: rgb(255, 255, 255);
  color: #dfcec1;
  padding: 50px 8px;
  height: 300px;
  border-bottom: 1px solid #8b6b52;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}
option .placeholder {
  color: #ff7e19;
}
.select select,
.select::after {
  grid-area: select;
}
.select:not(.select--multiple)::after {
  content: "";
  justify-self: end;
  width: 0.8em;
  height: 60px;
  background-color: var(--select-arrow);
  -webkit-clip-path: polygon(100% 0%, 0 0%, 50% 100%);
  clip-path: polygon(100% 0%, 0 0%, 50% 100%);
}
select[multiple] {
  padding-right: 0;
  height: 50px;
}
select[multiple] option {
  white-space: normal;
  outline-color: var(--select-focus);
}
@media (min-width: 760px) {
  select {
    height: 40px;
  }
}
