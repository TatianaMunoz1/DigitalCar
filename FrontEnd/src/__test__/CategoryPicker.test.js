import React, { useEffect } from "react";
import { shallow, mount, render, mock } from "enzyme";
import { Router } from "react-router-dom";
import CategoryPicker from "../components/CategoryPicker/CategoryPicker.js"

describe("category picker", () => {
    test("render category picker ", () => {
        const wrapper = shallow(<CategoryPicker/>)
        expect(wrapper).toMatchSnapshot();
    });
});