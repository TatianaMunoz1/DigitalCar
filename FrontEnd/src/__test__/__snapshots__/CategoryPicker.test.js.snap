// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category picker render category picker  1`] = `ShallowWrapper {}`;
